### ISSUE TITLE
Can't play local resources while offline

### ISSUE THREAD

**Issue by rdai** [arielfaur/ionic-audio#188]
Hi,

This is sort of a question/consulting help so I will be happy to donate to the project for the solution of this. 

What is working:
1.  I am loading a list of tracks from a RSS file, parse it and set all the src to remote/http locations
2. I then run a check against local files, if I find it locally I update the myTracks array, also tried to up _audioProvider.tracks array, not 100% sure which one is relevant. The reason I am doing it this instead of  try to check BEFORE adding to the myTracks is I really struggled with the asynchronous aspect, besides, I want to be able to "swap out" the src of a track once download is complete.
3. I can play local audio while online I can see this in the console. 

What is not working:
1.As soon as I turn off the cellular data in the emulator, the same file no longer plays, in my logs it appears to be trying to load the local resource but also checking the original, http src that I should have replaced. Somehow the original URL is stored somewhere and it is falling back to that instead of honoring what is in the myTracks array.

Console output when it is online for a previously downloaded file:
`vendor.js:62086 Loading track file:///data/user/0/com.myapp/files/file:___data_user_0_com_myapp_files_notreal_2018-03-02_hkff0290_mp3.mp3
2018-03-21 01:14:02.535 main.js:434 play event success undefined
2018-03-21 01:14:05.109 vendor.js:61857 Playing track file:///data/user/0/com.myapp/files/file:___data_user_0_com_`

Console output when offline, for the same file
`vendor.js:62091 GET http://traffic.libsyn.com/notreal/2018-03-02_hkff0290.mp3?dest-id=622857 net::ERR_INTERNET_DISCONNECTED
WebAudioTrack.play @ vendor.js:62091AudioTrackComponent.play @ vendor.js:77630AudioTrackPlayComponent.toggle @ vendor.js:77985(anonymous function) @ AudioTrackPlayComponent.ngfactory.js:41handleEvent @ vendor.js:13608handleEvent @ vendor.js:14335dispatchEvent @ vendor.js:10057(anonymous function) @ vendor.js:10671(anonymous function) @ vendor.js:35953t.invokeTask @ polyfills.js:3onInvokeTask @ vendor.js:4973t.invokeTask @ polyfills.js:3r.runTask @ polyfills.js:3e.invokeTask @ polyfills.js:3p @ polyfills.js:2v @ polyfills.js:2
2018-03-21 01:18:25.214 vendor.js:61849 Audio error => track file:///data/user/0/com.myapp/files/file:___data_user_0_com_myapp_files_notreal_2018-03-02_hkff0290_mp3.mp3 Event {isTrusted: true, type: "error", target: audio, currentTarget: audio, eventPhase: 2…}
2018-03-21 01:18:25.270 undefined:1 Uncaught (in promise) DOMException: Failed to load because no supported source was found.`

Things tried: I read in other issues often it is not finding the local path correctly so I have tried to append "this.file.dataDirectory" to the front of the file name, it doesn't make much difference: it still works when the device is online and it doesn't work when it is offline.

I noticed in the sample code there is a playSelectedTrack function, I tried to force it to play the right src again here that only resulted in TWO, "undefined:1 Uncaught (in promise) DOMException: Failed to load because no supported source was found." errors.

So to summarize, it appears overwritten the src only works when online? As soon as I am offline, it somehow resorts to the old http based URL and fails.

Just tried this on real device, in case it was a problem of web audio vs cordova. On phone, the play button is grayed out, so some how the canPlay check is failing somewhere. I probably need to dig more into that.

Any help would be greatly appreciated.

**Comment by arielfaur** [comment]
@rdai can you provide a link to a github project? Otherwise it is hard to see what's going on.

**Comment by rdai** [comment]
Ariel,

Thanks for the offer to help. I thought I give it one more shot and looking at the console logs in detail I was able to figure out that the _audioProvider object had a 'src' attribute on the base level, which I have been modifying to no effect, and another 'src' attribute in the _audioProvider['audio'] object and that is the one that mattered. 

So for any one else that this might help. To modify the src of a track after the initial loading (like after you downloaded the file), this format worked for me: **`this._audioProvider.tracks[findIdx]['audio']['src']`**  where 'findIdx' is an integer represent the position of the track that I found using an array search on the myTracks object.

This is what worked for me, you mileage might vary.
